### PR TITLE
JBTM-3880 Allow LRA users to provide base URI with configuration

### PR DESCRIPTION
**Pull request by xstefank**
https://issues.redhat.com/browse/JBTM-3880

!CORE !AS_TESTS !RTS !JACOCO !XTS !QA_JTA !QA_JTS_OPENJDKORB !PERFORMANCE LRA !DB_TESTS !MySQL !db2 !postgres !oracle

### CODE REVIEW

File: rts/lra/test/basic/src/test/java/io/narayana/lra/arquillian/LRACustomBaseURIIT.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    @Test
+    public void noLRATest() throws Exception {
+        try(Response response = client.target(baseURL.toURI())
+            .path(SIMPLE_PARTICIPANT_RESOURCE_PATH)
+            .path(START_LRA_PATH)
+            .request()
+            .get()) {
+
+            System.out.println(response.getHeaderString("Link"));
+            assertTrue("The base URI was not overridden by the configuration",
```
Comment (mmusgrov) [inline rts/lra/test/basic/src/test/java/io/narayana/lra/arquillian/LRACustomBaseURIIT.java:60]: This test is only testing the new `LinkCapturingFilter` whereas we need it to test that ServerLRAFilter is building the correct endpoints for the various participant endpoints.

**Review by mmusgrov — COMMENTED** [review]
FYI I haven't fully reviewed this PR yet because there is no test that the new config property does what we want with participant endpoints.

File: rts/lra/test/basic/src/test/java/io/narayana/lra/arquillian/LRACustomBaseURIIT.java (hunk 1 above)
Comment (xstefank) [reply rts/lra/test/basic/src/test/java/io/narayana/lra/arquillian/LRACustomBaseURIIT.java:60]: @mmusgrov this catches the Link header produced by the `ServerLRAFilter`. Please check the implementation.

**Review by mmusgrov — APPROVED** [review]
Okay so on further inspection I stand corrected - adding a test filter to validate the base URI is inspired and the PR does indeed validate that the changes to ServerLRAFilter correctly read the config to set the base URI, thanks for prompting me to take another look.

**Pull request merged**